### PR TITLE
Retry on GOAWAY error

### PR DESCRIPTION
**Pull request by TheRealHaoLiu**
https://github.com/ansible/awx/issues/13350

### CODE REVIEW

**Comment by TheRealHaoLiu** [comment]
test image with the change quay.io/haoliu/awx-ee:goaway

**Comment by TheRealHaoLiu** [comment]
test image updated with the latest change

File: pkg/workceptor/kubernetes.go [hunk 1]
```diff
@@ -630,18 +632,36 @@ func (kw *kubeUnit) runWorkUsingLogger() {
...
 
 						break
 					}
 
 					return
+				} else if _, ok := err.(http2.GoAwayError); ok {
+					// GOAWAY is sent by the server to indicate that the server is gracefully shutting down
+					// this happens if the kube API server we are connected to is being restarted or is shutting down
+					// for example during a cluster upgrade and rolling restart of the master node
+					kw.Debug(
```
Comment (fosterseth) [inline pkg/workceptor/kubernetes.go]: i'd say just make it `Info` level, should be relatively rare

File: pkg/workceptor/kubernetes.go [hunk 2]
```diff
@@ -630,18 +632,36 @@ func (kw *kubeUnit) runWorkUsingLogger() {
...
+						"Detected http2.GoAwayError for pod %s/%s. Will retry %d more times. Error: %s",
+						podNamespace,
+						podName,
+						remainingRetries,
+						err,
+					)
+					successfulWrite = false
```
Comment (fosterseth) [inline pkg/workceptor/kubernetes.go:658]: this section repeats twice. We could consider doing it once
```golang
if err == EOF {
  kw.debug(..)
else if err== GoAway {
  kw.info(..)
}
if err != nil {
  successfulWrite = false
  remainingRetries--
  if remainingRetries > 0 {
  ..
  }
}
```

**Review by fosterseth — APPROVED** [review]

**Review by CFSNM — APPROVED** [review]

**Comment by eranyo** [comment]
@TheRealHaoLiu  Many thanks for your help with this issue.

**Pull request merged**